django/cantusdb_project/main_app/views/views.py

```diff
@@ -959,6 +956,43 @@ def redirect_documents(request) -> HttpResponse:
     return redirect(new_path)
 
 
+def redirect_chant_list(request) -> HttpResponse:
+    source_id: str = request.GET.get("source")
+    if source_id is None:
+        # source parameter must be provided
+        raise BadRequest("Source parameter must be provided")
+
+    base_url: str = reverse("chant-list", args=[source_id])
+
+    # optional search params
+    feast_id: Optional[str] = request.GET.get("feast")
+    genre_id: Optional[str] = request.GET.get("genre")
+    folio: Optional[str] = request.GET.get("folio")
+    search_text: Optional[str] = request.GET.get("search_text")
+
+    d: dict = {
+        "feast": feast_id,
+        "genre": genre_id,
+        "folio": folio,
+        "search_text": search_text,
+    }
+    params: dict = {k: v for k, v in d.items() if v is not None}
+
+    query_string: str = urlencode(params)
+    url: str = f"{base_url}?{query_string}" if query_string else base_url
+
+    return redirect(url)
```

Comment:
this issues a temporary redirect by default, but I think this is intended to be permanent?

Reply:
Good catch

Reply:
Yes thanks!